gr-qtgui/python/qtgui/range.py.cmakein

```diff
@@ -51,6 +53,36 @@ class Range(object):
             val = 0
         return (val*self.step+self.min)
 
+class QEngValidator(Qt.QValidator):
+    def __init__(self, minimum, maximum, parent):
+        Qt.QValidator.__init__(self, parent)
+        self.min = minimum
+        self.max = maximum
+        self.parent = parent
+
+        self.re = r'^\d*(\.\d*)?((e\d*)|[EPTGMkmunpfa])?$'
+    def validate(self, s, pos):
+        try:
+            val=eng_notation.str_to_num(s)
+        except (IndexError, ValueError) as e:
+            if re.match(self.re,s):
+                self.parent.setStyleSheet("background-color: yellow;");
+                return (Qt.QValidator.Intermediate, s, pos)
+            else:
+                return (Qt.QValidator.Invalid, s, pos)
+
+        if self.max is not None and val > self.max:
+            self.parent.setStyleSheet("background-color: yellow;");
+
+        elif self.min is not None and val < self.min:
+            self.parent.setStyleSheet("background-color: yellow;");
+        else:
+
+            self.parent.setStyleSheet("background-color: white;");
+        return (Qt.QValidator.Acceptable, s, pos)
```

Comment:
add a CR/LF here

Reply:
made all the changes you requested in https://github.com/Sec42/gnuradio/commit/ed878a56428eb658be98f81033a0f5202001af9b